### ISSUE TITLE
Help-Wanted YAMLCreate.ps1

### ISSUE THREAD

**Issue by vedantmgoyal2009** [microsoft/winget-pkgs#21207]
Create issues related to the winget.exe client [here](https://github.com/microsoft/winget-cli/issues/new?template=Bug_Report.md)

### Brief description of your issue

![image](https://user-images.githubusercontent.com/83997633/125977259-0f275932-f9c6-4f32-9a4a-7e240899fdc2.png)

### Steps to reproduce
Can be seen in the screenshot above.

### Expected behavior
Should run normally.

### Actual behavior
Throwing error.

### Environment
Not related to winget.
```none
[winget --info]
Windows Package Manager version
Windows: Windows.Desktop version
Package: Microsoft.DesktopAppInstaller version

Any other software?
```

**Comment by quhxl** [comment]
Are you sure the folder `C:\Users\Bittu\Downloads\winget-pkgs\my-fork\manifests\k\kingsoft\kdocs` exists?

**Comment by vedantmgoyal2009** [comment]
Yes

**Comment by vedantmgoyal2009** [comment]
The issue came after #21172 has been merged.

**Comment by vedantmgoyal2009** [comment]
@OfficialEsco do you still have the old script, it worked fine for me.

**Comment by OfficialEsco** [comment]
This is a script error and user error.
Its looking for `kingsoft.kdocs.*.yaml` which obs does not exist when there is a singleton instead of `kingsoft.kdocs*.yaml`, aka it dosn't see `kingsoft.kdocs.yaml`
Its asking for the previous version number instead of the one you are trying to update to.

**Comment by vedantmgoyal2009** [comment]
But in the previous version of the script, nothing happened like this.

**Comment by OfficialEsco** [comment]
Because now the script is capable of updating a script in `Publisher/Software` when the folder structure is `Publisher/Software/Beta`
```
- Publisher
-- Software
--- Version 1
---- .yaml
--- Version 2
---- .yaml
--- Beta folder
---- Version 1
----- .yaml
```

Also it would really help to read instead of posting a issue/comment then deleting your issue/comment because you realized you did something wrong.
However it was a good catch since the MS Team is online and can merge it before the weekend.
Here is all the YamlCreate versions https://github.com/microsoft/winget-pkgs/commits/master/Tools/YamlCreate.ps1

**Comment by OfficialEsco** [comment]
Nevermind don't touch software with beta folders.. Someone smarter needs to fix it, like this @Trenly madman

**Comment by vedantmgoyal2009** [comment]
Worked fine now...

**Comment by Trenly** [comment]
> Nevermind don't touch software with beta folders.. Someone smarter needs to fix it, like this @Trenly madman

I think I ran into this issue with Google Chrome once. I'm guessing it has something to do with how the file/folder structure is extracted from the package name and version, but I'm not 100% sure.  

@OfficialEsco Would you mind creating a new issue separate from this one detailing the issues with the beta folders? I'd be happy to look at it, but don't want to co-mingle issues
